cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/service/http/BasicSystemHttpServiceContext.java

```diff
@@ -137,4 +141,9 @@ public void run(TxRunnable runnable) throws TransactionException {
       throw new IllegalArgumentException(String.format("Namespace '%s' does not exist", namespace), nfe);
     }
   }
+
+  @Override
+  public byte[] runTask(RunnableTaskRequest runnableTaskRequest) throws IOException {
+    return remoteTaskExecutor.runTask(runnableTaskRequest);
```

Comment:
The request should be rewritten to run the `SystemAppTask` instead of a task coming from the caller directly. This is because the RunnableTaskLauncher shouldn't contain special logic to distinguish system app task vs others (and based on having artifact or not is not a good way to distinguish it).

Reply:
Sounds good. Moved the logic to SystemAppTask.